### PR TITLE
Hw 3

### PR DESCRIPTION
**Pull request by Lemout17**
Added doctor and user pages
Added styled-components for pages

### CODE REVIEW

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "recommendations": ["jpoissonnier.vscode-styled-components"]
+}
```
Comment (codezavr) [inline .vscode/extensions.json]: We dont need your local files in our project. Remove and add it to .gitigrone

File: README.old.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# itrex-hw-3
+React based project
```
Comment (codezavr) [inline README.old.md]: What for do you need this file?

File: src/App.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import routes from './routes/routes'
+
+import PublicRoute from './routes/publicRoute'
+import PrivateRoute from './routes/privateRoute'
+
+import Loader from './components/preloader'
+
+// import RestoreView from './view/restoreView'
+// import ConfirmView from './view/confirmView'
+// import NotFound from './view/notFound'
```
Comment (codezavr) [inline src/App.js]: Remove unused code.

File: src/components/containers/section/section.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import imgMobile from '../../../img/bg-mobile@x2.jpg'
+import imgTablet from '../../../img/bg-tablet@x2.jpg'
+import imgDesktop from '../../../img/bg-desktop@x2.jpg'
+
+const SectionWrapper = styled.section`
```
Comment (codezavr) [inline src/components/containers/section/section.js]: You can move these type of constructions into separate files to make component clearer. Example SectionWrapper.styled.tsx and after use it in you component via import.

File: src/components/form/confirmForm/confirmForm.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import { NavLink } from 'react-router-dom'
+import styled from 'styled-components'
+import Title from '../../title'
+import Text from '../formStyles/text'
+import Link from '../formStyles/link'
+import routes from '../../../routes/routes'
+
+const Container = styled.div`
```
Comment (codezavr) [inline src/components/form/confirmForm/confirmForm.js]: Move into separate file

File: src/components/form/formStyles/button.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import styled from 'styled-components'
+import arrowRightSvg from '../../../svg/angle-right-b.svg'
+
+const Button = styled.button`
```
Comment (codezavr) [inline src/components/form/formStyles/button.js]: Move all styled things into separate fileы across the project.

File: src/components/mainDoctor/emptyWrapper/emptyWrapper.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import { EmptyWrapper, EmptyIcon, EmptyText } from '../styles'
+
+const EmptyState = () => {
+  return (
+    <EmptyWrapper>
+      <EmptyIcon />
+      <EmptyText>
+        You have no patients yet. <br />
```
Comment (codezavr) [inline src/components/mainDoctor/emptyWrapper/emptyWrapper.js]: Prevent to use BRs in you project, better solution to use styled spans. This case is fine, but when you use alot of BRs its a bad practice.

File: src/redux/auth/authActions.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+const registerRequest = createAction('auth/registerRequest')
+const registerSuccess = createAction('auth/registerSuccess')
+const registerError = createAction('auth/registerError')
+
+const loginRequest = createAction('auth/loginRequest')
+const loginSuccess = createAction('auth/loginSuccess')
+const loginError = createAction('auth/loginError')
+
+const googleRequest = createAction('auth/googleRequest')
```
Comment (codezavr) [inline src/redux/auth/authActions.js]: What for do you need this action?

File: src/view/signInView/index.js [hunk 9]
```diff
@@ -0,0 +1 @@
+export { default } from './signInView'
```
Comment (codezavr) [inline src/view/signInView/index.js]: What for do you need to create index.js and export default every time?

Is it for const SignInView = lazy(() =>
  import('./view/signInView' /* webpackChunkName: "signin-page" */)
)???

File: src/components/containers/main/index.js [hunk 10]
```diff
@@ -0,0 +1 @@
+export { default } from './main'
```
Comment (codezavr) [inline src/components/containers/main/index.js]: Why dont you use just main, what is the reason of creation new files?

File: src/components/form/signInForm/signInForm.js [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+    }),
+    onSubmit: () => dispatch(authActions.registerSuccess()),
+  })
+
+  return (
+    <>
+      <Form onSubmit={formik.handleSubmit}>
+        <Title text="Sign In" />
+
+        <div>
```
Comment (codezavr) [inline src/components/form/signInForm/signInForm.js:60]: Could be removed.

File: src/components/form/signUpForm/signUpForm.js [hunk 12]
```diff
@@ -0,0 +1,215 @@
...
+  const dispatch = useDispatch()
+
+  const handleToggle = (e) => {
+    if (e.target.id === 'togglePass1') {
+      setTogglePass1(!togglePass1)
+    }
+
+    if (e.target.id === 'togglePass2') {
+      setTogglePass2(!togglePass2)
+    }
```
Comment (codezavr) [inline src/components/form/signUpForm/signUpForm.js]: Duplicates, create function that will handle both cases.

File: src/components/form/signUpForm/signUpForm.js [hunk 13]
```diff
@@ -0,0 +1,215 @@
...
+    }),
+    onSubmit: () => dispatch(authActions.registerSuccess()),
+  })
+
+  return (
+    <>
+      <Form onSubmit={formik.handleSubmit}>
+        <Title text="Sign Up" />
+
+        <div>
```
Comment (codezavr) [inline src/components/form/signUpForm/signUpForm.js:82]: Unused wrapper, remove

File: src/components/header/userAvatar/userAvatar.js [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+  width: 15px;
+  height: 15px;
+
+  border: 3px solid var(--doctor-bg);
+  border-radius: 50%;
+  background-color: var(--online-color);
+`
+
+const UserAvatar = ({ avatar }) => {
+  console.log(avatar)
```
Comment (codezavr) [inline src/components/header/userAvatar/userAvatar.js]: Remove

File: src/components/mainDoctor/emptyWrapper/emptyWrapper.js [hunk 15]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import { EmptyWrapper, EmptyIcon, EmptyText } from '../styles'
+
+const EmptyState = () => {
+  return (
+    <EmptyWrapper>
```
Comment (codezavr) [inline src/components/mainDoctor/emptyWrapper/emptyWrapper.js]: But ints not empty, it has content in it.

File: src/components/mainPatient/mainPatient.js [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+const MainPatient = () => {
+  const [state, setstate] = useState([data])
+  const width = useWindowWidth()
+
+  return (
+    <MainWrapper>
+      <ButtonWrapper>
+        <Button white>Profile</Button>
+        <Button>Appointments</Button>
+        {width >= 1280 && (
```
Comment (codezavr) [inline src/components/mainPatient/mainPatient.js]: Kinda wierd practice, to use it in template, probably you can resolve this using STYLES.

File: src/components/mainPatient/mainPatient.js [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+          </Button>
+        )}
+      </ButtonWrapper>
+
+      <FilterWrapper>
+        <Title doctor text="My appointments" />
+        <FilterButton patient />
+      </FilterWrapper>
+
+      {state.length <= 0 ? <EmptyState /> : <AppointmentsList data={state} />}
```
Comment (codezavr) [inline src/components/mainPatient/mainPatient.js]: How state can be less then 0 if you have emty array [], empty array has 0 length.
Also you can simplify this: !state.length

File: src/index.css [hunk 18]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+img {
+  display: block;
+  width: 100%;
+  height: auto;
+}
+
+.preloader {
+  margin-top: 100px;
```
Comment (codezavr) [inline src/index.css]: Bad practice to use margin-top.

**Review by codezavr — CHANGES_REQUESTED** [review]

**Review by matthewlaz — COMMENTED** [review]
I've heard that it is good practice to name styled elements like <StyledButton /> <StyledWrapper /> so that “Styled” before the styled element.  This is not a remark, this is more a question and I share my thoughts

File: src/components/mainDoctor/emptyWrapper/emptyWrapper.js (hunk 7 above)
Comment (Lemout17) [reply src/components/mainDoctor/emptyWrapper/emptyWrapper.js]: Okey  but I have no idea how to split them using css

File: src/redux/auth/authActions.js (hunk 8 above)
Comment (Lemout17) [reply src/redux/auth/authActions.js]: I add this for future logining when we take token from backend and save it in local storage

File: src/view/signInView/index.js (hunk 9 above)
Comment (Lemout17) [reply src/view/signInView/index.js]: it's for shorter imports (good practice)

File: src/components/form/signInForm/signInForm.js (hunk 11 above)
Comment (Lemout17) [reply src/components/form/signInForm/signInForm.js:60]: if i remove this div css rules (&:nth-child(n)::before and last-child)  become unsuported.

File: src/components/form/signUpForm/signUpForm.js (hunk 13 above)
Comment (Lemout17) [reply src/components/form/signUpForm/signUpForm.js:82]: this div used for css rules (&:nth-child(n)::before and last-child)
I can't remove it.

File: src/components/containers/main/index.js (hunk 10 above)
Comment (Lemout17) [reply src/components/containers/main/index.js]: okay I remove this component and use  main tag

File: src/components/mainDoctor/emptyWrapper/emptyWrapper.js (hunk 15 above)
Comment (Lemout17) [reply src/components/mainDoctor/emptyWrapper/emptyWrapper.js]: I will rename this component for better understanding

File: src/components/mainPatient/mainPatient.js (hunk 16 above)
Comment (Lemout17) [reply src/components/mainPatient/mainPatient.js]: I will use styles then

File: src/components/form/formStyles/button.js (hunk 6 above)
Comment (Lemout17) [reply src/components/form/formStyles/button.js]: I didn't quite understand what I should do
They already separated

**Review by codezavr — APPROVED** [review]

File: src/components/mainDoctor/emptyWrapper/emptyWrapper.js (hunk 7 above)
Comment (codezavr) [reply src/components/mainDoctor/emptyWrapper/emptyWrapper.js]: Usually we create two divs / spans and use it to divide text with margins

**Review by codezavr — APPROVED** [review]
Well done, now you merge it

**Pull request merged**